registry_store.go

```diff
@@ -79,14 +80,19 @@ type RegistryStoreOptions struct {
 	// MaxManifestBytes limits the maximum size of the fetched manifest bytes.
 	// If less than or equal to zero, a default value (4 MiB) will be used.
 	MaxManifestBytes int64
+
+	// IncludeTagBasedCosignSignature enables fetching cosign signatures with
+	// the tag format when listing referrers.
+	IncludeTagBasedCosignSignature bool
```

Comment:
We can have a better name like `AllowCosignTag`? It's called _Cosign tag-based discovery_.

References:
- https://github.com/sigstore/cosign/blob/v2.5.2/specs/SIGNATURE_SPEC.md#tag-based-discovery

You need to reference that reference in your code comment.